pkg/network/usm/postgres_monitor_test.go

```diff
@@ -97,11 +117,32 @@ func (s *postgresProtocolParsingSuite) TestDecoding() {
 	}
 }
 
+// waitForPostgresServer verifies that the postgres server is up and running.
+// It tries to connect to the server until it succeeds or the timeout is reached.
+// We need that function (and cannot relay on the RunServer method) as the target regex is being logged a couple os
+// milliseconds before the server is actually ready to accept connections.
+func waitForPostgresServer(t *testing.T, serverAddress string, enableTLS bool) {
+	pgClient, err := postgres.NewPGXClient(postgres.ConnectionOptions{
+		ServerAddress: serverAddress,
+		EnableTLS:     enableTLS,
+	})
+	require.NoError(t, err)
+	defer pgClient.Close()
+	require.Eventually(t, func() bool {
+		return pgClient.Ping() == nil
+	}, 5*time.Second, 100*time.Millisecond, "couldn't connect to postgres server")
+}
+
 func testDecoding(t *testing.T, isTLS bool) {
```

Comment:
I do feel we currently have some limitations in the code which we do not cover with tests, maybe lets think about some tests to test our limits as done in http2 to validate those limts.

Reply:
can you share examples?

Reply:
Actually it does make sense to do so with the row test